### PR TITLE
Add lang attribute to <html> tag for accessibility improvement.

### LINKED ISSUE #9630: Links do not have a discernible name

### LINKED ISSUE #9634: element does not have a [lang] attribute

### PR DESCRIPTION
**Pull request by Jash2606**
Closes #9634

### Description

This PR adds a `lang` attribute to the `<html>` tag to specify the primary language of the document. This attribute is crucial for accessibility, as it helps screen readers and other assistive technologies interpret the language correctly, ensuring proper pronunciation and comprehension for users. The `lang` attribute has been set to `"en"` for English, as this is the primary language of the content.

### Testing

1. Open the modified HTML document in a browser.
2. Inspect the `<html>` tag to confirm the presence of the `lang="en"` attribute.
3. Use a screen reader or accessibility tool to verify that it recognizes the language of the document as English.

### Stakeholders
@cdrini

### CODE REVIEW

**Comment by Jash2606** [comment]
Hey @scottbarnes @cdrini , I have updated the lang attribute as mentioned in issue #9634.
Please review the changes and let me know if any further updates are needed.
Thank you!

**Comment by scottbarnes** [comment]
Thanks for this, @Jash2606. I noticed #9634 has the label `Needs: Detail`, so I started a conversation there in the hope we could figure out how to proceed with this issue. As things stand, I am not convinced this PR is fully responsive to the issue raised. However, more discussion may help tease out what, if anything, should be done.

**Comment by scottbarnes** [comment]
Thanks for this, @Jash2606. Just a heads up that you probably want to add this change to `openlibrary/openlibrary/templates/site/head.html`.

I just skimmed the code and I can't actually tell if this particular sitemap function is being used, but I imagine if the proposed change makes sense in `head.html`, it probably makes sense in `sitemap.py` as well.

But either way, please test out the changes you've made, e.g., as I did with `curl`, and show them in the PR so it's easier to review.

**Comment by Jash2606** [comment]
> Thanks for this, @Jash2606. Just a heads up that you probably want to add this change to `openlibrary/openlibrary/templates/site/head.html`.
> 
> I just skimmed the code and I can't actually tell if this particular sitemap function is being used, but I imagine if the proposed change makes sense in `head.html`, it probably makes sense in `sitemap.py` as well.
> 
> But either way, please test out the changes you've made, e.g., as I did with `curl`, and show them in the PR so it's easier to review.

Hey @scottbarnes ,

Changes done as suggested for head.html also !

**Comment by rebecca-shoptaw** [comment]
Hi @Jash2606! I had a look at the info in the issue, and this seems like a good approach to me. 

I'd second what @scottbarnes said and suggest you test via `curl` to confirm it's working as expected. Once you get your Docker container up and running you can try the following:
```
curl -s http://localhost:8080 | head -n 5 
```
The site should now default to `lang="en"` and look like this:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
```

You'll also want to confirm that the language is correctly switched if the bot selects a language, by running:
```
curl -s -H "Accept-Language: es" http://localhost:8080 | head -n 5
```
The site should still switch to `lang="es"` and look like this:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es">
```
Let me know how it goes! I'll try testing on my end as well, but it would be good to document your own test results in the PR. 🙂

**Comment by rebecca-shoptaw** [comment]
@Jash2606 **Update:** I pulled the PR up on GitPod, and it looks like the code is going to need a slight reword!

Here's what it looks like in the dev tools currently:
New html lang screenshot

It looks like this is just happening because you forgot the curly braces in the move from `sitemap` to `head`; once those are added it should work currently, pending further `curl` testing. 🙂

**Comment by Jash2606** [comment]
> @Jash2606 **Update:** I pulled the PR up on GitPod, and it looks like the code is going to need a slight reword!
> 
> Here's what it looks like in the dev tools currently: New html lang screenshot
> 
> It looks like this is just happening because you forgot the curly braces in the move from `sitemap` to `head`; once those are added it should work correctly. 🙂
> 
> Tried adding in the curly braces on my end, and confirmed: ✅ `curl` with no language header returns `lang="en"` ✅ `curl` with a Spanish language header returns `lang="es"`

Hey @rebecca-shoptaw ,

Updated Syntax changes , Let me know if any further changes required.

**Comment by rebecca-shoptaw** [comment]
@Jash2606 Great, thank you, and thanks for your great work on this! 

I'll test once more to double check, then mark the PR ready to merge. Also, for your next PR I'd recommend having a look at the [Git Cheat Sheet](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) -- you'll find it much easier to keep things up to date and avoid manual merges if you create a new branch for each PR, and rebase off the master branch if needed. 🙂

**Review by rebecca-shoptaw — APPROVED** [review]
@Jash2606 Great, thank you, and thanks for your great work on this! 

Tested via GitPod:
✅ `curl` with no specified language defaults to `lang="en"`
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
<head>
```
✅ `curl` with a language specified uses that language
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es">
<head>
```

Also, for your next PR I'd recommend having a look at the [Git Cheat Sheet](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) -- you'll find it much easier to keep things up to date and avoid manual merges if you create a new branch for each PR, and rebase off the master branch if needed. 🙂

File: openlibrary/data/sitemap.py [hunk 1]
```diff
@@ -48,7 +48,7 @@
     """$def with (page)
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
     "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
-<html xmlns="http://www.w3.org/1999/xhtml">
+<html xmlns="http://www.w3.org/1999/xhtml" lang="${get_lang() or 'en'}">
```
Comment (cdrini) [inline openlibrary/data/sitemap.py:51]:
```suggestion
<html xmlns="http://www.w3.org/1999/xhtml">
```

File: openlibrary/templates/site/head.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 $def with (page)
 
 <!DOCTYPE html>
-<html xmlns="http://www.w3.org/1999/xhtml" lang="$get_lang()">
+<html xmlns="http://www.w3.org/1999/xhtml" lang="${get_lang() or 'en'}">
```
Comment (cdrini) [inline openlibrary/templates/site/head.html:4]:
```suggestion
<html xmlns="http://www.w3.org/1999/xhtml" lang="$(get_lang() or 'en')">
```

**Pull request merged**